### DISCUSSION TITLE
Vips supports RAW images by fall back on ImageMagick, but I don't see this behaviour in GoVips?

### DISCUSSION THREAD

**Discussion by enricoribelli** [davidbyttow/govips#403] [Q&A]
Hello team! As per title of the question, I am having troubles processing RAW images through GoVips. These get recognised as TIFF instead of `imageType == MAGICK` . I have tested through Vips itself that these images can get processed, so Vips works by itself and that should be normal because Vips falls back to using ImageMagick for raw images. But whenever I try to load and process them through GoVips I get:

`failed to process images: failed to process raw image: failed to get image: MissingRequired: TIFF directory is missing required \"ImageLength\" field\nvips__tiff_openin_source: unable to open source for input\n\nStack:\ngoroutine 1 [running]:\nruntime/debug.Stack()\n\t/opt/homebrew/Cellar/go/1.21.3/libexec/src/runtime/debug/stack.go:24 +0x64\ngithub.com/davidbyttow/govips/v2/vips.handleVipsError() ....etc...`

Is this behaviour not supported in GoVips? I can't find a statement anywhere about this unfortunately. Thank you!

**Comment by tonimelisma** [comment]
Hey @enricoribelli, sure smells like a bug to me. How comfortable are you with golang, you want to hazard a try to debug and submit a PR?

**Reply by enricoribelli** [reply]
Will give it a try! Golang is fine.. problem is when it steps into the C area:)
